cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -124,5 +139,31 @@ cudf::size_type distinct_count(table_view const& input,
                                null_equality nulls_equal    = null_equality::EQUAL,
                                rmm::cuda_stream_view stream = rmm::cuda_stream_default);
 
+/**
+ * @brief Generate a gather map that refers to distinct rows of the input table.
+ *
+ * Given an input table, this function returns a gather map such that gathering the input table
+ * using this map will produce an equivalent result (i.e., the same set of rows regarding of
+ * order) as the result of calling to `cudf::distinct`. As such, this function requires the same
+ * parameters as `cudf::distinct`.
```

Comment:
```suggestion
 * Given an `input` table_view, each row index is copied to the output array in an
 * unspecified order if the corresponding row of `keys` columns is distinct (no other
 * equivalent row exists in the table).
```
I feel "gather map" is a too specific term thus we may want to avoid it in the doc. The explanation of why the input parameters are the same as `distinct` seems unnecessary.